spigot-impl/src/main/java/me/aurium/beetle/spigot/SpigotOptions.java

```diff
@@ -0,0 +1,104 @@
+package me.aurium.beetle.spigot;
+
+import me.aurium.beetle.api.BeetleOptions;
+import me.aurium.beetle.api.task.sync.BlockingBehavior;
+import me.aurium.beetle.defaults.task.sync.TimerSleepBehavior;
+
+import java.util.Objects;
+import java.util.concurrent.Executor;
+
+public class SpigotOptions implements BeetleOptions {
+
+    private final boolean isDebug;
+    private final boolean usePlatformExecutor;
+    private final boolean useThisThread;
+
+    private final Executor platformExecutor;
+    private final Thread mainThread;
+    private final BlockingBehavior blockingBehavior;
+
+    public SpigotOptions(boolean isDebug, boolean usePlatformExecutor, boolean useThisThread, Executor platformExecutor, Thread mainThread, BlockingBehavior blockingBehavior) {
```

Comment:
Neither this constructor nor the builder check if any of these parameters are null. What happens when an API user forgets to specify the platformExecutor?

Reply:
ignored by latest commit